### DISCUSSION TITLE
Delay

### DISCUSSION THREAD

**Discussion by HenrySetUp** [NightscoutFoundation/xDrip#3161] [Q&A]
We have just upgraded my sons phone to a new google pixel 8 from the 4a, we have reloaded xdrip to work as a companion app alongside CamAps to then use Dexcom follow on mine and my wife’s iOS phone. This was all working fine before the upgrade. 

Now everything is working but the Glucose readings keep dropping and are taking a long time to upload. It can be 2 mins then over 10mins and then it says it has lost the signal on xdrip. 

We have set everting up the same as was before. Have we missed a setting or has something changed. 

We have been hunting for a solution for nearly a month with no avail.

**Comment by Navid200** [comment]
Can you make it clear if it is the master (Pixel 8) that has the delay and drops, or your followers?

**Reply by HenrySetUp** [reply]
Yes the master phone with the Camaps and Xdrip is the Pixel 8

**Reply by Navid200** [reply]
Yes, that is clear in your original post already.
What is not clear still is if the delay and drops you are reporting occur on the master or the follower.
That's a big difference and we cannot help until you clarify that.

**Reply by HenrySetUp** [reply]
So it drops on the Pixel 8 phone which is the master and the phone that has the Xdrip app.  So it has readings for 2 mins then it will leave a gap for more than 10mins it may take a reading and then comes up in the top left hand corner on the xdrip app saying signal missed. 

Once it says the signal has been missed nothing shows on the Dexcom follow.

**Reply by Navid200** [reply]
OK, perfect.  Then, since the problem is on the master, let's forget about the followers then for now.

When there are missed readings in xDrip, does CamAPS show notifications?

What is the status of your son's old phone?  Is he still turning it on?  Does it still have CamAPS installed on it?  If yes, that's very likely the cause of the problem.  You should not have CamAPS on two different phones getting readings from the same transmitter.

**Reply by HenrySetUp** [reply]
Yes so the Camaps still showed notifications and readings at 5min intervals, the xdrip is 2 mins on one reading the gap can be 8-10mins plus then it says signal loss. 

The old mobile has been erased and turned off so this can’t be the issue. 

Can I share a picture on here.

**Reply by Navid200** [reply]
Of course you can share a picture.  While editing, click on the bottom border, which offers to upload attachments.

**Reply by Navid200** [reply]
Please delete the unsuccessful uploads.
And please don't upload anything from the follower.  We already talked about this.  The follower is irrelevant at this point until the master problem is resolved.  You will only be cluttering this thread by uploading or entering anything about the followers.

**Reply by HenrySetUp** [reply]
![Uploading IMG_9761.jpeg…]()

**Reply by HenrySetUp** [reply]
Ok so the image I am trying to send is of the xdrip screen with the big gaps and the signal missed

**Reply by Navid200** [reply]
This post explains how to attach an image: https://github.com/NightscoutFoundation/xDrip/discussions/1900

What is important is to see what version of xDrip you have.
And if you have followed all of this: https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html

**Reply by HenrySetUp** [reply]
Ok so  version of xdrip is 23ccbd1-2023.07.16

all the setting and setup has been done about 50 times over the past week to try and get over this problem we have come across.

**Reply by Navid200** [reply]
Install the latest Nightly version of xDrip instead: https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.10.24

Have you disabled battery optimization of xDrip in Android settings?

**Reply by HenrySetUp** [reply]
He had gone to bed now, but I will upload this in the morning and update you on the result.  I am I’m pretty sure we have already done this one and it was the same, I will give it another go though. 

I’m also sure we did the battery option but I will 100% check these in the morning. 

Thank you for your time this Evening

**Reply by Navid200** [reply]
No problem

You may also want to enable Bluetooth (the last heading here: https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html) to see if missing points may be resolved.

**Reply by HenrySetUp** [reply]
So this morning we uninstalled xdrip, downloaded and set up the recommended install in your previous message. Checked all the settings etc and it still does the same thing. Xdip takes a reading at 2 mins then doesn’t change and then says signal missed in the top left corner in red, then it won’t take another reading for 10mins sometimes up to 18-20mins. 

We have uninstalled and set this up now I think probably 20 times. All was working fine on the old phone with think an install that was done 12months + ago.

**Reply by jamorham** [reply]
Please update to the latest nightly if not done already and while using xDrip you should get a prompt to disable battery optimizations make sure that you allow that because it sounds a lot like a powersaving issue on the phone.

**Reply by HenrySetUp** [reply]
We have updated to the latest version again yesterday and we have defiantly change the battery optimisation to unrestricted and it still does the same thing. 

It seems no one can give me a solution.
